### PR TITLE
316 anyone in an organization can invite other people to join that organization 

### LINKED ISSUE #316: Anyone in an organization can invite other people to join that organization
Once you are a part of an organization, you're able to invite anyone else to join that organization. 

You can do this with a time-limited sharing link.

Joining a Lightdash org will be by invite only (i.e. you can't just try to login with your work email address and get added to your org by default....baby steps)

## Tasks

- [x] Create invite link table
- [x] Create backend for `POST /invites`
- [x] Create UI for creating invites

### PR DESCRIPTION
**Pull request by owlas**
Closes #316

### CODE REVIEW

File: packages/frontend/src/hooks/useInviteLink.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                const [title, ...rest] = error1.error.message.split('\n');
+                showToastError({
+                    title,
+                    subtitle: rest.join('\n'),
+                });
+            },
+            onSuccess: (data) => {
+                showToastSuccess({
+                    title: 'Created new invite link',
+                    subtitle: `Expires on ${data.expiresAt.toDateString()}`,
```
Comment (ZeRego) [inline packages/frontend/src/hooks/useInviteLink.tsx:43]: Should we display dates/time in the same format everywhere in the app ? 
We have a `formatTimestamp()` util function that displays in `YYYY-MM-DD HH:mm:ss`

File: packages/backend/src/analytics/LightdashAnalytics.ts [hunk 2]
```diff
@@ -21,43 +21,29 @@ type Group = {
...
+type TrackSimpleEvent = BaseTrack & {
+    event:
+        | 'user.created'
+        | 'user.logged_in'
+        | 'user.updated'
+        | 'password.updated'
+        | 'organization.created'
+        | 'organization.updated'
+        | 'invite_link.created'
+        | 'query.executed';
```
Comment (ZeRego) [inline packages/backend/src/analytics/LightdashAnalytics.ts:33]: I like the new naming convention :)

**Comment by ZeRego** [comment]
@owlas I think the user should have a way to revoke the invite link in case they share it somewhere public by mistake. 

When they open the organization settings we could fetch and display the existing and valid invite link. There would also be a button to revoke that invite. Once revoked we would display the button to create a new one. 
To prevent multiple valid invites, in `UserService.createOrganizationInviteLink`, we would delete all existing invites before creating a new one.

**Comment by owlas** [comment]
@ZeRego this sounds pretty sensible. At the moment we mint a new token each time you request an invite link and it expires after 3 days.

This stops you sharing a link that will expire very soon and also keeps a relatively short time that a link is active.

There is definitely a scenario where a user accidentally shares this invite link, knowingly, and within the expiry time. Then they would want to revoke the token. We can move this to a separate issue?

**Review by ZeRego — APPROVED** [review]
LGTM 🎉 😄 

We can address the revoke invite flow in a separate PR.

**Review by TuringLovesDeathMetal — APPROVED** [review]

**Pull request merged**